### ISSUE TITLE
populate IntegrationSyntax

### ISSUE THREAD

**Issue by i-walker** [arrow-kt/arrow-meta#663]
There are a couple of missing extension to complement cli-ide integration as proposed in #652 
The steps are similar for each extension X in the bucketlist below:
1) Create an extension in `arrow.meta.ide.phases.integration` with the same name as X [here](https://github.com/arrow-kt/arrow-meta/blob/9a8529614c78fd33652b03c28110bedfc7434019/idea-plugin/src/main/kotlin/arrow/meta/ide/phases/integration/integration.kt#L7-L6) are a couple of examples 
```kotlin
package arrow.meta.ide.phases.integration
//...

interface X : ExtensionPhase {
  fun `meaningfull function name`(project : Project): arrow.meta.<...>.X
}
```

2) Add an extension function that creates X in `arrow.meta.ide.dsl.integration.IntegrationSyntax` [examples](https://github.com/arrow-kt/arrow-meta/blob/c26453a5b69b81c66ac3c940590506529aefd025/idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/integration/IntegrationSyntax.kt#L18)

3) Register X in `IdeInternalRegistry` [examples](https://github.com/arrow-kt/arrow-meta/blob/882ea6cf86cea51be305c722feb4f913d08f08f7/idea-plugin/src/main/kotlin/arrow/meta/ide/internal/registry/IdeInternalRegistry.kt#L91)

Feel free to ask in this issue, if there are any blockers or questions 😄 

Bucketlist:
- [ ] arrow.meta.phases.resolve.synthetics.SyntheticScopeProvider
- [ ] arrow.meta.phases.codegen.ir.IRGeneration
- [ ] arrow.meta.phases.resolve.DeclarationAttributeAlterer
- [ ] arrow.meta.phases.codegen.asm.Codegen
- [ ] arrow.meta.phases.codegen.asm.ClassBuilder
- [ ] arrow.meta.phases.analysis.AnalysisHandler
- [ ] arrow.meta.phases.config.StorageComponentContainer
- [ ] arrow.meta.phases.analysis.PreprocessedVirtualFileFactory
- [ ] arrow.meta.phases.analysis.ExtraImports
- [ ] arrow.meta.phases.config.Config
- [ ] arrow.meta.phases.analysis.CollectAdditionalSources

**Comment by Starmel** [comment]
I take it

**Comment by i-walker** [comment]
resolved in #727 
Thanks to @Starmel
